DOCUMENTATION.md

```diff
@@ -1370,529 +1395,202 @@ The following are example configurations.
+In the unisocket client mode, the client will only connect to one of the configured addresses. This single member will behave as a gateway to the other members. For any operation requested from the client, it will redirect the request to the relevant member and return the response back to the client returned from this member.
 
-> `caPath`, `keyPath` and `certPath` define the file path to the respective file that stores such information.
+## 7.3. Handling Failures
 
-```json
-{
-    "network": {
-        "ssl": {
-            "enabled": true,
-            "factory": {
-                "exportedName": "BasicSSLOptionsFactory",
-                "properties": {
-                    "caPath": "ca.pem",
-                    "keyPath": "key.pem",
-                    "certPath": "cert.pem",
-                    "rejectUnauthorized": false
-                }
-            }
-        }
-    }
-}
-```
+There are two main failure cases you should be aware of. Below sections explain these and the configurations you can perform to achieve proper behavior.
 
-If these options are not enough for your application, you may write your own options factory and instruct the client
-to get the options from it, as explained below.
+### 7.3.1. Handling Client Connection Failure
 
-**Writing an `SSLOptionsFactory`**
+While the client is trying to connect initially to one of the members in the `ClientNetworkConfig.addressList`, all the members might not be available.
+Instead of giving up, throwing an error and stopping the client, the client retries to connect as configured which is
+described in the [Configuring Client Connection Retry](#61-configuring-client-connection-retry) section.
 
-In order to use the full range of options provided to [`tls.connect`](https://nodejs.org/api/tls.html#tls_tls_connect_options_callback) of Node.js, you may write your own factory object.
+The client executes each operation through the already established connection to the cluster. If this connection(s) disconnects or drops, the client will try to reconnect as configured.
 
-An example configuration is shown below.
+### 7.3.2. Handling Retry-able Operation Failure
 
-```json
-{
-    "network": {
-        "ssl": {
-            "enabled": true,
-            "factory": {
-                "path": "my_factory.js",
-                "exportedName": "SSLFactory",
-                "properties": {
-                    "caPath": "ca.pem",
-                    "keyPath": "key.pem",
-                    "certPath": "cert.pem",
-                    "keepOrder": true
-                }
-            }
-        }
-    }
-}
-```
+While sending the requests to the related members, the operations can fail due to various reasons. Read-only operations are retried by default. If you want to enable retrying for the other operations, you can set the `redoOperation` to `true`. See the [Enabling Redo Operation section](#53-enabling-redo-operation).
```

Comment:
nit: `While sending requests to cluster members, the operations may fail due to various reasons.` sounds a bit better to me. WDYT?